File: node-web/src/lib/retention.test.ts
import { describe, it, expect } from 'vitest'
import { retentionExpiresAt, RETENTION_DAYS } from './retention'

describe('retentionExpiresAt', () => {
  it('returns a Date exactly 180 days after the given instant', () => {
    const now = Date.UTC(2026, 0, 1)
    const result = retentionExpiresAt(now)
    expect(RETENTION_DAYS).toBe(180)
    expect(result.getTime() - now).toBe(180 * 24 * 60 * 60 * 1000)
  })

  it('defaults to Date.now() when no instant is passed', () => {
    const before = Date.now()
    const result = retentionExpiresAt()
    const after = Date.now()
    expect(result.getTime()).toBeGreaterThanOrEqual(before + 180 * 24 * 60 * 60 * 1000)
    expect(result.getTime()).toBeLessThanOrEqual(after + 180 * 24 * 60 * 60 * 1000)
  })
})

File: test/core/retention_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:soter_recovery/core/retention.dart';

void main() {
  group('retentionExpiresAt', () {
    test('returns exactly 180 days after the given instant', () {
      final now = DateTime.utc(2026, 1, 1);
      final result = retentionExpiresAt(now);
      expect(retentionDays, 180);
      expect(result.difference(now), const Duration(days: 180));
    });

    test('normalizes to UTC', () {
      final now = DateTime(2026, 1, 1);
      final result = retentionExpiresAt(now);
      expect(result.isUtc, isTrue);
    });

    test('defaults to DateTime.now() when no instant is passed', () {
      final before = DateTime.now().toUtc();
      final result = retentionExpiresAt();
      final after = DateTime.now().toUtc();
      expect(
        result.isAfter(before.add(const Duration(days: 180)).subtract(const Duration(seconds: 1))),
        isTrue,
      );
      expect(
        result.isBefore(after.add(const Duration(days: 180)).add(const Duration(seconds: 1))),
        isTrue,
      );
    });
  });
}
